qiskit/circuit/quantumcircuit.py

```diff
@@ -137,7 +141,7 @@ class QuantumCircuit:
     header = "OPENQASM 2.0;"
     extension_lib = "include \"qelib1.inc\";"
 
-    def __init__(self, *regs, name=None, global_phase=0):
+    def __init__(self, *regs, name=None, global_phase=0, **metadata):
```

Comment:
Do we want to add it during construction, or only assign it as an attribute later? If we keep it here maybe it would be good to keep this an explicity kwarg `metadata=None` rather than arbitrary kwargs.

Reply:
+1 to `metadata=None`

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/5270/commits/26036dc51ee57c5d4a7dd6bd866d73bf07233cb7